Makefile

```diff
@@ -165,12 +165,13 @@ test-unit: $(SETUP_ENVTEST) #HELP Run the unit tests
                 $(UNIT_TEST_DIRS) \
                 -test.gocoverdir=$(ROOT_DIR)/coverage/unit
 
-image-registry: ## Setup in-cluster image registry
-	./hack/test/image-registry.sh $(E2E_REGISTRY_NAMESPACE) $(E2E_REGISTRY_NAME)
-
-build-push-e2e-catalog: ## Build the testdata catalog used for e2e tests and push it to the image registry
-	./hack/test/build-push-e2e-catalog.sh $(E2E_REGISTRY_NAMESPACE) $(LOCAL_REGISTRY_HOST)/$(E2E_TEST_CATALOG_V1)
-	./hack/test/build-push-e2e-catalog.sh $(E2E_REGISTRY_NAMESPACE) $(LOCAL_REGISTRY_HOST)/$(E2E_TEST_CATALOG_V2)
+E2E_REGISTRY_IMAGE=localhost/e2e-test-registry:devel
+image-registry: ## Build the testdata catalog used for e2e tests and push it to the image registry
+	go build $(GO_BUILD_FLAGS) -tags '$(GO_BUILD_TAGS)' -ldflags '$(GO_BUILD_LDFLAGS)' -gcflags '$(GO_BUILD_GCFLAGS)' -asmflags '$(GO_BUILD_ASMFLAGS)' -o ./testdata/registry/bin/registry ./testdata/registry/registry.go
```

Comment:
```suggestion
	GOOS=linux GOARCH=amd64 go build $(GO_BUILD_FLAGS) -tags '$(GO_BUILD_TAGS)' -ldflags '$(GO_BUILD_LDFLAGS)' -gcflags '$(GO_BUILD_GCFLAGS)' -asmflags '$(GO_BUILD_ASMFLAGS)' -o ./testdata/registry/bin/registry ./testdata/registry/registry.go
```

Reply:
Since the `test-e2e` target has a dependency on the `image-registry` target, the `GOOS` and `GOARCH` variables should be set from the `test-e2e` target, and don't need to be set explicitly here. This way, you can build `image-registry` locally, and get the correct (local) images when no running the e2e.

Reply:
Can we use `GOOS=$(go env GOOS) GOARCH=$(go env GOARCH)` instead? This would make the target compatible with both macOS and Linux environments. Setting GOOS=linux GOARCH=amd64 explicitly restricts it to those environments, meaning we can’t test or run it locally on other systems like macOS.

Reply:
Hi @tmshort and @bentito,

Would it make more sense to use GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) instead?

This approach would make the target compatible with both macOS and Linux environments, allowing us to run it locally as well. If we explicitly set GOARCH=amd64, for example, it won’t run on macOS systems with a different architecture, where the expected result would be darwin/arm64. Is targeting linux/amd64 specifically what we're aiming for here?

Reply:
yep. current changes take this into account.

Reply:
Hi @tmshort,

> Not sure that's the right approach. Doing the export should work when doing the image-target, and setting the variable equal to the go env ought to be a noop.

I’m not sure if I fully follow your comment here. The export has fixed values as well:

```
image-registry: export GOOS=linux
image-registry: export GOARCH=amd64
```

Either way, this suggests **fixed** values for go build.

So, how would `GOOS=linux GOARCH=amd64 go build` build a binary for darwin/arm64 (my local env, for example)?

Reply:
Right, but doing this:
```
GOOS=$(go env GOOS)
```
Is basically setting the `GOOS` variable to the environment value that the `go` command already has available to it. It's equivalent to doing:
```
GOOS=${GOOS}
```
Which is a no-op.

Reply:
@camilamacedo86  I didn't answer your question. 

> So, how would `GOOS=linux GOARCH=amd64 go build` build a binary for darwin/arm64 (my local env, for example)?

It wouldn't. In _this_ particular case, it's building executables for the docker container, which have to be linux/amd64. These two executables are never expected to run locally.

My point is that doing `GOOS=$(go env GODS` is pointless.

Reply:
This build is specifically for the `image-registry` image to be run in `kind`; comparing to how we build the `operator-controller`'s `manager` binary, we use the same `GOOS=linux GOARCH=amd64` ENV when building (`make run` uses `docker-build` which then uses `build-linux`). I'm just following that same convention.

If you feel it's important, what we could potentially do is split up the `image-registry` target so that `GOOS=linux GOARCH=amd64` is only set when actually building the image. But I think this is unnecessary or at least non-critical enough to address in a follow-up.

Reply:
> These two executables are never expected to run locally.

If that is the case, (we never run it locally and would not make sense run it locally) then I am OK with fixed values 👍